aws_test.go

```diff
@@ -1,3 +1,5 @@
+// +build !docker
+
```

Comment:
Do the aws tests not run with the docker engine?

Reply:
All tests that use the generic-worker payload won't run in docker builds since it uses the docker-worker payload.

Reply:
The plan we had was to add support to generic-worker for running tasks inside a docker container (bugs [1502335](https://bugzil.la/1502335),  [1502337](https://bugzil.la/1502337),  [1502339](https://bugzil.la/1502339),  [1502348](https://bugzil.la/1502348),  [1502351](https://bugzil.la/1502351),  [1502352](https://bugzil.la/1502352)), and _then_ providing a compatibility layer so that docker-worker style payloads would be supported (bug [1502354](https://bugzil.la/1502354)). It sounds like at the moment with these changes, generic-worker style payloads won't work when running a task in docker, which would mean it we don't have a trivial path to align on payload format in future. It also makes the changes more complicated, because all code that touched the payload has to be adapted to work with a different payload format. The idea behind the compatibility layer was that we have an isolated function that converts from one payload format to the other, and all existing code continues to work with the existing data structures, tasks can be interpreted in either format, and we have a simple path for migrating to a standard format.